### PR TITLE
Migrate to support capabilities, in favor of polymorphism

### PR DESCRIPTION
**Pull request by mikenachbaur-okta**
Note: This removes the `@dynamicMemberLookup` experimental features, since it was confusing the compiler, and wasn't adding much value to the DX.

### CODE REVIEW

File: Sources/OktaIdx/IDXSocialAuthRemediation.swift [hunk 1]
```diff
@@ -11,7 +11,7 @@
 //
 
 import Foundation
-
+/*
```
Comment (JayNewstrom) [inline Sources/OktaIdx/IDXSocialAuthRemediation.swift:14]: Should this file be removed?

File: Sources/OktaIdx/Capabilities/IDXCapability.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import Foundation
+
+public protocol AuthenticatorCapability {
+}
+
```
Comment (JayNewstrom) [inline Sources/OktaIdx/Capabilities/IDXCapability.swift:17]: Worth adding some documentation for these new types?

**Review by JayNewstrom — APPROVED** [review]
Overall this looks like it's going down the same path as the Kotlin SDK, and looks good. But might be worth having another iOS dev take a look too.

File: Sources/OktaIdx/IDXSocialAuthRemediation.swift (hunk 1 above)
Comment (mikenachbaur-okta) [reply Sources/OktaIdx/IDXSocialAuthRemediation.swift:14]: Nice catch. 😉

File: Sources/OktaIdx/Capabilities/IDXCapability.swift (hunk 2 above)
Comment (mikenachbaur-okta) [reply Sources/OktaIdx/Capabilities/IDXCapability.swift:17]: Yes, I'll be adding more docs this afternoon. Dotting I's, crossing T's, etc. 

Largely these protocols are to define which capabilities are applicable to authenticators vs remediations, but in my mind, everything that's public must be documented.

**Review by oleggnidets-okta — APPROVED** [review]

**Pull request merged**